index.src.html

```diff
@@ -4144,6 +4147,43 @@
 : <dfn event for=ImageTrack>change</dfn>
 :: Fired at the {{ImageTrack}} when the {{ImageTrack/frameCount}} is altered.
 
+Resource Reclamation{#resource-reclamation}
+==============================================
+
+When resources are constrained, a User Agent may proactively reclaim codecs.
+This is particularly true in the case where hardware codecs are limited, and
+shared accross all web pages.
+
+A User Agent may only [=reclaim a codec=] that is either an
+[=inactive codec=], a [=background codec=], or both. A User Agent MUST NOT
+reclaim a codec that is both active and in the foreground, e.g any codec that is
+neither an [=inactive codec=] nor a [=background codec=].
+
+An <dfn>inactive codec</dfn> is a codec that has not received a call to
+`encode()` or `decode()` in the past `3 seconds`, and has not called its
+`output()` callback in the past `3 seconds`.
+
+A <dfn>background codec</dfn> is a codec whose {{ownerDocument}}'s
+{{Document/hidden}} attribute is {{hidden|"hidden"}}.
+
+To <dfn>reclaim a codec</dfn>, a User Agent must run the appropriate
+close algorithm (amongst [=Close AudioDecoder=], [=Close AudioEncoder=],
+[=Close VideoDecoder=] and [=Close VideoEncoder=]) with a {{QuotaExceededError}}
+{{DOMException}}.
+
+When selecting which codec to reclaim, the User Agent should consider the
+following criteria:
+    - What type of resource is the codec holding on to? User Agents should
+        reclaim codecs that will alleviate the pressure of scarce resources.
+    - How disruptive might reclaiming the codec be? User Agents should consider
+        whether the codec is a [=background codec=], an [=inactive codec=], or
+        both. Codecs that are both are least likely disrupt to user experience
+        when reclaimed. Otherwise, User Agents might prefer reclaiming codecs
+        that are [=background codec=]s, before [=inactive codec=]s in
+        foreground pages.
+    - Any other heuristics, determined by the User Agent. For example,
+        heuristics might prioritize reclaiming the oldest codecs, the ones with
+        the least recent activity, or the least frequent activity.
```

Comment:
This is going to be a web compat nightmare.

Reply:
In my initial read of this I agreed, but then I re-read the intro and felt better.

> A User Agent may only reclaim a codec that is either an inactive codec, a background codec, or both. A User Agent MUST NOT reclaim a codec that is both active and in the foreground, e.g any codec that is neither an inactive codec nor a background codec.

This caused me to read the highlighted sentence as purely informational  description (perhaps it should be a note) of the narrow flexibility afforded by the above rule and how UAs might prioritize codecs for reclamation. For instance, the UA may choose to reclaim only one of two inactive codecs if one codec was holding onto a more precious resource than the other. The UA may also choose to reclaim both innactive codecs or neither. But this sentence does not allow the UA to reclaim an _active foreground_ codec.

Reply:
This needs to be specced a bit more tightly, https://github.com/w3c/webcodecs/issues/121#issuecomment-875750242 is my latest thinking, transcribed by @chcunningham.